pyvista/utilities/cells.py

```diff
@@ -54,14 +87,16 @@ def __init__(self, cells=None, n_cells=None, deep=False):
 
     def _set_cells(self, cells, n_cells, deep):
         vtk_idarr, cells = numpy_to_idarr(cells, deep=deep, return_ind=True)
-        # get number of cells if none
+
+        # get number of cells if None.  This is quite a performance
+        # bottleneck and we can consider adding a warning.  Good
+        # candidate for Cython or JIT compilation
         if n_cells is None:
             if cells.ndim == 1:
-                c = 0
-                n_cells = 0
-                while c < cells.size:
-                    c += cells[c] + 1
-                    n_cells += 1
+                if sys.version_info.minor > 6:
+                    n_cells = ncells_from_cells_py36(cells)
+                else:  # About 20% slower
```

Comment:
Yes.  Though our CI does test on 3.6.

Reply:
I figure that's the case because you quickly noticed my `n_cells` refactor breaking. Still, for local coverage reports it's probably better with the pragma.